Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -466,27 +466,47 @@ internal async Task<BoolResult> RestoreCheckpointAsync(OperationContext context,
 
         private async Task<BoolResult> ProcessStateAsync(OperationContext context, bool inline)
         {
-            try
+            var result = await processStateCoreAsync();

```

Comment:
>processStateCoreAsync [](start = 31, length = 21)

I think it would be cleaner (with added benefit of logging) to wrap the method body in PerformOperation and then just set _postInitializationTask to BoolResult.SuccessTask in case where it succeeds (no checks on current task status). We might also consider tracking the time since the last successful result for the purposes of flagging repeated checkpoint restore or create checkpoint failures for too long of a period of time.